File: common/src/main/java/org/sqlunet/browser/HomeFragment.java
/*
 * Copyright (c) 2023. Bernard Bou
 */

package org.sqlunet.browser;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

import org.sqlunet.browser.common.R;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.navigation.fragment.NavHostFragment;

/**
 * Home fragment
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public class HomeFragment extends Fragment
{
	/**
	 * Mandatory empty constructor for the fragment manager to instantiate the fragment (e.g. upon screen orientation changes).
	 */
	public HomeFragment()
	{
	}

	@Nullable
	@Override
	public View onCreateView(@NonNull final LayoutInflater inflater, final ViewGroup container, @Nullable final Bundle savedInstanceState)
	{
		final View view = inflater.inflate(R.layout.fragment_home, container, false);

		FloatingActionButton fab = view.findViewById(R.id.fab);
		fab.setOnClickListener((v) -> {
			final FragmentActivity activity = requireActivity();
			final NavHostFragment navHostFragment = (NavHostFragment) activity.getSupportFragmentManager().findFragmentById(R.id.nav_host_fragment);
			assert navHostFragment != null;
			navHostFragment.getNavController().navigate(R.id.nav_search_browse);
		});

		return view;
	}

	@Override
	public void onResume()
	{
		super.onResume();

		final AppCompatActivity activity = (AppCompatActivity) requireActivity();
		final ActionBar actionBar = activity.getSupportActionBar();
		assert actionBar != null;
		actionBar.setCustomView(null);
		actionBar.setBackgroundDrawable(null);
	}
}

File: common/src/main/java/org/sqlunet/browser/HomeFragment.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.browser

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.navigation.NavOptions
import androidx.navigation.findNavController
import com.google.android.material.floatingactionbutton.FloatingActionButton
import org.sqlunet.browser.common.R

/**
 * Home fragment
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
open class HomeFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_home, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // fab
        val fab = view.findViewById<FloatingActionButton>(R.id.fab)
        fab.setOnClickListener {
            navigateToBrowse()
        }

        // toolbar
        var toolbar = requireActivity().findViewById<Toolbar>(R.id.toolbar)

            val menuProvider = object : MenuProvider {

                override fun onCreateMenu(menu: Menu, inflater: MenuInflater) {
                    menu.clear()
                    inflater.inflate(R.menu.main_safedata, menu)
                 }

                override fun onMenuItemSelected(item: MenuItem): Boolean {
                    return when (item.itemId) {
                        R.id.search -> {
                            navigateToBrowse()
                            true
                        }
                        else -> false
                    }
                }
            }

            requireActivity().addMenuProvider(
                menuProvider,
                viewLifecycleOwner,
                Lifecycle.State.RESUMED
            )
        }
    }

    override fun onResume() {
        super.onResume()
        val activity = requireActivity() as AppCompatActivity
        val actionBar = activity.supportActionBar!!
        actionBar.customView = null
        actionBar.setBackgroundDrawable(null)
    }

    private fun navigateToBrowse() {
        val navController = requireActivity().findNavController(R.id.nav_host_fragment)
        navController.navigate(
            R.id.nav_search_browse,
            null,
            NavOptions.Builder()
                .setPopUpTo(R.id.nav_home, true)  // clear back stack up to nav_home inclusive
                .build()
        )
    }
}
